### ISSUE TITLE
Custom admonition icons don't work when defining theme.palette

### ISSUE THREAD

**Issue by locomoco28** [squidfunk/mkdocs-material#3987]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When trying to specify a custom icon for admonitions, the icons are not replaced when the `mkdocs.yml` also has `theme.palette` defined.

### Expected behaviour

After commenting out the `theme.palette` key, we can see the expected behavior

![image](https://user-images.githubusercontent.com/32169493/172003351-c114c449-1e2c-4d0c-8294-03cd0f9fd500.png)

### Actual behaviour

The icon does not get replaced when specifying the `theme.palette` key

![image](https://user-images.githubusercontent.com/32169493/172003395-0a06a078-6af4-46ee-bdec-f5e1b4636b6e.png)

### Steps to reproduce

Deploy the following example page using the mkdocs.yml specified [below](#Configuration)

```markdown
<!-- docs/index.md -->
!!! note
    This is an example note
```

### Package versions

- Python: `3.8.13`
- MkDocs: `1.3.0`
- Material: `8.3.1`

### Configuration

```yaml
# mkdocs.yml
site_name: "Example mkdocs material"

theme:
  name: material
  palette: # <- this is the culprit, commenting it out fixes the issue, but we use need it for dark/bright mode configuration
    scheme: default
  icon:
    admonition:
      note: octicons/alert-24
    logo: material/cloud

markdown_extensions:
  - admonition
  - pymdownx.details
  - pymdownx.superfences
```

### System information

Locally using Win 10 and also in a newly created Gitpod.io workspace (cloud VS Code IDE)

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in 7998dbf. The problem was that we used SASS' selector extension feature to allow defining `[data-md-color-scheme=default]` on nested content, so you can put any element into dark or light mode. Since we extended `:root`, all definitions where carried over to `[data-md-color-scheme=default]`, including those of admonition icons. The linked commit moves definitions into a placeholder to keep things separate.

**Comment by squidfunk** [comment]
Released as part of 8.3.2.

**Comment by locomoco28** [comment]
Thanks for the quick fix, very much appreciated <3
